src/tribler-core/tribler_core/upgrade/version_manager.py

```diff
@@ -189,3 +190,52 @@ def fork_state_directory_if_necessary(root_state_dir, code_version):
 
     if version_history:
         version_history.update(code_version)
+
+
+def get_disposable_state_directories(root_state_dir, code_version, skip_last_version=True):
+    version_history, _, src_dir, tgt_dir, last_version = should_fork_state_directory(root_state_dir, code_version)
+
+    if src_dir is not None:
+        # If versioned state directory exists
+        if src_dir != root_state_dir:
+            skip_dirs = [src_dir, tgt_dir]
+            skip_versions = [last_version]
+
+            # Get the second last version if available and add it to the dirs to save
+            second_last_version = version_history.get_last_upgradable_version(root_state_dir, last_version)
```

Comment:
Why do this `skip_versions` hack instead of just forming a full list of versions and using Python slicing to select every one but the last two?

Reply:
1. `skip_versions` is used to give flexibility to choose different version from Settings/Data page.
2. If only last two version are selected, it does not clean up non-version directories eg. pre 7.4 directories
3. Full list of version should be sorted separately in the order of installation based on version history, additional logic

Reply:
I would expect _choosing_ the version to delete from the Data page to be completely independent from the process _listing_ the versions to choose from

>Full list of version should be sorted separately in the order of installation based on version history, additional logic

I was not clear enough in my request, my fault. By "the full list of versions" I mean "the ordered list of versions". By "ordered" I mean "ordered by last usage time, last usage goes last".

So, the expected logic is:
1. Get the list of state directories from the directory listing A)
2. Get the ordered list of versions from the version history file (B) 
3. Filter list B to only contain entries from A
4. Get `B[:-2]` and show it to the user. Let the user form the list C of versions to delete.
5. Delete every version in list C.

Reply:
I'm not sure that the `B[:-2]` logic ("we should suggest to delete everything except two last directories") is correct. If we currently have versions:
- 8.3
- 8.4
- **8.5 (currently running version)**
- 8.6
- 8.7
what should we delete? In my opinion, definitely not `[8.3, 8.4, 8.5]`. I think the correct answer is `[8.3, 8.6, 8.7]`.